src/Submission/ReduxMapFiles.js

```diff
@@ -3,16 +3,22 @@ import MapFiles from './MapFiles';
 import { fetchWithCreds } from '../actions';
 import { indexdPath } from '../localconf';
 
-const fetchUnmappedFiles = user => dispatch => fetchWithCreds({
-  path: `${indexdPath}index?acl=null&uploader=${user}`,
+const FETCH_LIMIT = 1024;
+
+const fetchUnmappedFiles = (user, total, start) => dispatch => fetchWithCreds({
+  path: `${indexdPath}index?acl=null&uploader=${user}&start=${start}&limit=${FETCH_LIMIT}`,
   method: 'GET',
 }).then(
   ({ status, data }) => {
     switch (status) {
     case 200:
+      total = total.concat(data.records);
+      if (data.records.length === FETCH_LIMIT) {
+        return dispatch(fetchUnmappedFiles(user, total, data.records[FETCH_LIMIT - 1].did));
```

Comment:
Not sure, I might be wrong but will this last record `data.records[FETCH_LIMIT - 1]` download from backend in the next fetch? Or there's some other code already remove the duplicated records...?

Reply:
I'm not sure honestly... the way pagination works with IndexD is via DID. So to paginate you have to take the last file did and pass it in as a starting point. I'm assuming the starting point is exclusive?

Reply:
🤔then what about the very first record of the first fetch, will it also be excluded?